core/controllers/email_dashboard_test.py

```diff
@@ -290,19 +331,29 @@ def test_that_valid_exceptions_are_raised(self):
                 query_models[0].query_status,
                 feconf.USER_QUERY_STATUS_COMPLETED)
 
-        # Check that exception is raised for unauthorized user.
-        self.login(self.USER_A_EMAIL)
-        with self.assertRaisesRegexp(Exception, '401 Unauthorized'):
-            self.get_html_response(
-                '/emaildashboardresult/%s' % query_models[0].id)
-        self.logout()
-
         # Check that exception is raised if current user is not submitter of
         # that query.
         self.login(self.NEW_SUBMITTER_EMAIL)
         with self.assertRaisesRegexp(Exception, '401 Unauthorized'):
             self.get_html_response(
                 '/emaildashboardresult/%s' % query_models[0].id)
+        # FIXME: Not catching unauthorized call for post requests.
+        with self.assertRaisesRegexp(Exception, '401 Unauthorized'):
+            self.post_json(
+                '/emaildashboardresult/%s' % query_models[0].id, {
+                    'data': {
+                        'email_subject': 'subject',
+                        'email_body': 'body',
+                        'max_recipients': None,
+                        'email_intent': 'bulk_email_marketing'}})
+        with self.assertRaisesRegexp(Exception, '401 Unauthorized'):
```

Comment:
"user_id is not an authorized user for this query". Isn't this the error message that is raised? Or does the post_json raise another error? I think you have got the right idea here. I'm not very familiar with this controller file, and not sure what the query_model is. I'll get back to you on this.